### PR TITLE
navigate to Search page after fully closing modal on keyboard shortcut

### PR DESCRIPTION
**Pull request by situchan**
### Details
Navigate to new page only after modal is fully closed, which means focus is ready to be transferred.

### Fixed Issues
$ https://github.com/Expensify/App/issues/15276
PROPOSAL: https://github.com/Expensify/App/issues/15276#issuecomment-1475404755

### Tests
Same as QA step

- [ ] Verify that no errors appear in the JS console

### Offline tests
Same as QA step

### QA Steps
1. Login with any account on web/desktop app
2. Open FAB menu
3. Press CTRL + K
4. Verify that FAB menu closes, search page opens and input field is focused automatically
5. Close search page
6. Right click any chat message to show context menu
7. Press CTRL + Shift + K
8. Verify that context menu closes, new group page opens and input field is focused automatically

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/108292595/226825475-fcb3571e-0de3-4a78-8400-d9dbce790cb1.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/108292595/226824965-2d15f8f6-685f-4d93-a38e-28b1c8f16f20.mov

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/108292595/226824903-4fac1efb-8b64-48b7-aa10-bf27fa3dbbcb.mp4

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/108292595/226825429-4939ab3c-cb40-4da4-95fa-be4b1a4c5d32.mov

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/108292595/226824813-a9fb57fb-b8cf-415f-bc87-ccfeec82946a.mp4

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/108292595/226824761-88c7c28b-989d-490a-804e-4b13016c53ac.mov

</details>

### CODE REVIEW

**Comment by situchan** [comment]
> @situchan The input doesn't seem to be focused on the EmojiPicker navigating using `Ctrl + k` can you take a look at it?

@getusha It works for me. Which browser / version is it?

**Comment by situchan** [comment]
> The input doesn't seem to be focused on the EmojiPicker navigating using Ctrl + k can you take a look at it?

We discussed [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1679492463884599?thread_ts=1679489813.001499&cid=C01GTK53T8Q), found root cause and solution.
I pushed fix.

@getusha please pull again from https://github.com/situchan/Expensify/tree/fix/15276-search-focus-shortcut

**Comment by situchan** [comment]
> @situchan Will test it again, thanks! in the meantime i suggested to add comment to explain the `onClose` above, it can be confusing since we're calling the callback inside `close` function. can we do that?

I don't see any suggestions. Did you submit?

**Comment by situchan** [comment]
image

**Comment by situchan** [comment]
@getusha good suggestions. pushed updates. please check

**Comment by situchan** [comment]
@getusha thanks for testing. If you've done review, let's move to next step. So ping @Santhosh-Sellavel ready for review

**Comment by situchan** [comment]
@Santhosh-Sellavel kindly bump for review

**Comment by Santhosh-Sellavel** [comment]
> @getusha thanks for testing. If you've done review, let's move to next step. So ping @Santhosh-Sellavel ready for review

I was just waiting for the ping from @getusha as you said above!

**Comment by situchan** [comment]
Oh, I thought 👍 from @getusha meant done for review. Sorry for confusion.
@getusha please comment here so we could get the PR reviewed and merged asap.

File: src/libs/actions/Modal.js [hunk 1]
```diff
@@ -12,9 +13,27 @@ function setCloseModal(onClose) {
...
-function close() {
-    if (!closeModal) { return; }
-    closeModal();
+/**
+ * Close modal in other parts of the app
+ *
+ * @param {Function} [onModalHideCallback]
+ * @param {Boolean} isNavigating
+ */
+function close(onModalHideCallback, isNavigating = true) {
```
Comment (Santhosh-Sellavel) [inline src/libs/actions/Modal.js:22]: `onModalHideCallback` can be rename as `onModalCloseCallback`

**Review by Santhosh-Sellavel — CHANGES_REQUESTED** [review]
@situchan 

Seems to be working well for most cases.

Steps
1. Press CMD+K will open Search Page focus is set ✅ 
2. Now Press CMD+Shift+K will New Group is set ✅ 
3. Now Press CMD+K  will open Search Page focus is not set ❌ 

https://user-images.githubusercontent.com/85645967/227319829-fddec81f-6163-4670-b3ec-820c68ae9ed5.mov

cc: @hayata-suenaga

**Comment by Santhosh-Sellavel** [comment]
@hayata-suenaga Let me know what should we do about this one https://github.com/Expensify/App/pull/16381#pullrequestreview-1355326714, thanks!

**Comment by situchan** [comment]
> Now Press CMD+K will open Search Page focus is not set ❌

Adding my cents here:
This is from default behavior of react-navigation.
Search page is already mounted so upon search page shortcut key on New group page, it just closes and go back to existing navigation route.
You can see that transition animation from New group page to Search page is reversed, which is same as when click back button.

**Comment by Santhosh-Sellavel** [comment]
Yeah, I got that too.

**Comment by hayata-suenaga** [comment]
@Santhosh-Sellavel yes I agree that this issue seems to be out of scope of the original issue.

**Comment by situchan** [comment]
@Santhosh-Sellavel do you have more feedback for us to address?

**Comment by Santhosh-Sellavel** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/85645967/227654301-81783151-d744-46d6-8fc3-1255e2806eea.mov

</details>

<details>
<summary>Mobile Web - Chrome & Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/85645967/227654118-b67ffe41-e2f0-4f32-b835-d046ed86d3a0.mov

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/85645967/227654278-8f60ab03-0e0e-44b5-8c5c-7b8d21c2d1e0.mov

</details>

<details>
<summary>iOS & Android</summary>

https://user-images.githubusercontent.com/85645967/227654067-bc9cd8c0-943f-4d95-a6ce-fc640fcb255e.mov

</details>

**Review by Santhosh-Sellavel — APPROVED** [review]
LGTM tests well. All you @hayata-suenaga!

**Review by getusha — APPROVED** [review]

**Review by hayata-suenaga — APPROVED** [review]
LGTM!

**Pull request merged**